### PR TITLE
Security fixes: privilege escalation control, run as non root, drop capabilities

### PR DESCRIPTION
**Pull request by malayparida2000**
run containers as allowPrivilegeEscalation:false
    resolves- https://bugzilla.redhat.com/show_bug.cgi?id=2114847

drop all default capabilities for containers
    resolves- https://bugzilla.redhat.com/show_bug.cgi?id=2114844

run pods as runAsNonRoot: true
    resolves- https://bugzilla.redhat.com/show_bug.cgi?id=2114840

### CODE REVIEW

File: deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -2756,6 +2756,13 @@ spec:
                   initialDelaySeconds: 5
                   periodSeconds: 10
                 resources: {}
+                securityContext:
```
Comment (iamniting) [inline deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml:2759]: Can you pls check why these changes are not getting generated for the metrics exporter?

**Review by iamniting — CHANGES_REQUESTED** [review]

File: deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml (hunk 1 above)
Comment (malayparida2000) [reply deploy/bundle/manifests/ocs-operator.clusterserviceversion.yaml:2759]: I am not sure, But I just found out metrics exporter actually needs privileged access. Why it needs has been explained [here](https://github.com/red-hat-storage/ocs-operator/pull/1638/commits/6d081d662abfdf722f98fadbb63511e008f42b1c). So I am reverting the changes for the metrics-exporter.

**Review by umangachapagain — COMMENTED** [review]
Squash the changes into one commit.
lgtm

**Comment by malayparida2000** [comment]
> Squash the changes into one commit. lgtm

Had to keep separate commits as these are referring to 3 different BZs, Would squashing them all to one be okay?

**Review by iamniting — APPROVED** [review]
/lgtm
